src/Model/Contact.php

```diff
@@ -94,6 +94,210 @@ class Contact extends BaseObject
+			if (empty($pcid)) {
+				return [];
+			}
+			$ucid = $contact['id'];
+		} else {
+			$pcid = $contact['id'];
+			$ucid = Contact::getIdForURL($contact['url'], $uid, true);
+		}
+
+		return ['public' => $pcid, 'user' => $ucid];
+	}
+
+	/**
+	 * @brief Block contact id for user id
+	 *
+	 * @param int     $cid     Either public contact id or user's contact id
+	 * @param int     $uid     User ID
+	 * @param boolean $blocked Is the contact blocked or unblocked?
+	 */
+	public static function setBlocked($cid, $uid, $blocked)
+	{
+		$cdata = self::getPublicAndUserContacID($cid, $uid);
+		if (empty($cdata)) {
+			return;
+		}
+
+		if ($cdata['user'] != 0) {
+			DBA::update('contact', ['blocked' => $blocked], ['id' => $cdata['user'], 'pending' => false]);
+		}
+
+		DBA::update('user-contact', ['blocked' => $blocked], ['cid' => $cdata['public'], 'uid' => $uid], true);
+	}
+
+	/**
+	 * @brief Returns "block" state for contact id and user id
+	 *
+	 * @param int $cid Either public contact id or user's contact id
+	 * @param int $uid User ID
+	 *
+	 * @return boolean is the contact id blocked for the given user?
+	 */
+	public static function getBlocked($cid, $uid)
+	{
+		$cdata = self::getPublicAndUserContacID($cid, $uid);
+		if (empty($cdata)) {
+			return;
+		}
+
+		$public_blocked = false;
+
+		if (!empty($cdata['public'])) {
+			$public_contact = DBA::selectFirst('user-contact', ['blocked'], ['cid' => $cdata['public'], 'uid' => $uid]);
+			if (DBA::isResult($public_contact)) {
+				$public_blocked = $public_contact['blocked'];
+			}
+		}
+
+		$user_blocked = $public_blocked;
+
+		if (!empty($cdata['user'])) {
+			$user_contact = DBA::selectFirst('contact', ['blocked'], ['id' => $cdata['user'], 'pending' => false]);
+			if (DBA::isResult($user_contact)) {
+				$user_blocked = $user_contact['blocked'];
+			}
+		}
+
+		if ($user_blocked != $public_blocked) {
+			DBA::update('user-contact', ['blocked' => $user_blocked], ['cid' => $cdata['public'], 'uid' => $uid], true);
+		}
+
+		return $user_blocked;
+	}
+
+	/**
+	 * @brief Ignore contact id for user id
+	 *
+	 * @param int     $cid     Either public contact id or user's contact id
+	 * @param int     $uid     User ID
+	 * @param boolean $ignored Is the contact ignored or unignored?
+	 */
+	public static function setIgnored($cid, $uid, $ignored)
+	{
+		$cdata = self::getPublicAndUserContacID($cid, $uid);
+		if (empty($cdata)) {
+			return;
+		}
+
+		if ($cdata['user'] != 0) {
+			DBA::update('contact', ['readonly' => $ignored], ['id' => $cdata['user'], 'pending' => false]);
+		}
+
+		DBA::update('user-contact', ['ignored' => $ignored], ['cid' => $cdata['public'], 'uid' => $uid], true);
+	}
+
+	/**
+	 * @brief Returns "ignore" state for contact id and user id
+	 *
+	 * @param int $cid Either public contact id or user's contact id
+	 * @param int $uid User ID
+	 *
+	 * @return boolean is the contact id ignored for the given user?
+	 */
+	public static function getIgnored($cid, $uid)
```

Comment:
Since this function returns a boolean and depends on a user Id, please rename to `isIgnoredByUser`.